osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyReverseArrow.cs

```diff
@@ -68,11 +71,42 @@ private void onHitObjectApplied(DrawableHitObject drawableObject)
                                                     .OverlayElementContainer.Add(proxy);
         }
 
+        private void updateStateTransforms(DrawableHitObject hitObject, ArmedState state)
+        {
+            const double move_out_duration = 35;
+            const double move_in_duration = 250;
+            const double total = 300;
+
+            switch (state)
+            {
+                case ArmedState.Idle:
+                    if (shouldRotate)
+                    {
+                        InternalChild.ScaleTo(1.3f, move_out_duration)
+                                     .RotateTo(5.625f)
+                                     .Then()
+                                     .ScaleTo(1f, move_in_duration)
+                                     .RotateTo(-5.625f, move_in_duration)
+                                     .Loop(total - (move_in_duration + move_out_duration));
+                    }
+                    else
+                    {
+                        InternalChild.ScaleTo(1.3f)
+                                     .Then()
+                                     .ScaleTo(1f, move_in_duration)
+                                     .Loop(total - (move_in_duration + move_out_duration));
+                    }
+
+                    break;
+            }
+        }
+
         protected override void Dispose(bool isDisposing)
         {
             base.Dispose(isDisposing);
-            if (drawableHitObject != null)
```

Comment:
In all three classes, I think we still want these null-checks to guard against cases of possible disposal before dependency injection. Will need to use `.IsNotNull()` rather than standard checks, but yeah.